src/dependency-manager.ts

```diff
@@ -87,4 +104,112 @@ export class DependencyManager {
         //     console.debug('error:' + err)
         // })
     }
+
+    private async downloadDepsRecurisively(rootPath: string, useNpm: boolean): Promise<void> {
+        try {
+            const names = await fs.readdir(rootPath);
+            for (const name of names) {
+                const childPath = path.resolve(rootPath, name);
+                const stat = await fs.stat(childPath);
+                if (stat.isDirectory()) {
+                    if (name === 'node_modules') {
+                        return;
+                    }
+                    const packageFile = path.resolve(childPath, 'package.json');
+                    const lockFile = path.resolve(childPath, 'yarn.lock');
+
+                    if (fs.existsSync(packageFile)) {
+                        if (useNpm) {
+                            this.runNpm(childPath);
+                        } else { // use npm
+                            if (fs.existsSync(lockFile)) {
+                                await this.downloadDeps(childPath);
+                            } else {
+                                this.logger.info('No yarn.lock file existed under: ' + childPath + ', consider enabled npm downloading.');
+                            }
+                        }
+                    } else {
+                        await this.downloadDepsRecurisively(childPath, useNpm);
+                    }
+                }
+            }
+        } catch (e) {
+            this.logger.error(e);
+        }
+    }
+
+    public async downloadFile(url: URL, dest: string): Promise<void> {
+        return new Promise((resolve, reject) => {
+            const lib = url.protocol.startsWith('https') ? https : http;
+            const request = lib.get(url, (response: http.IncomingMessage) => {
```

Comment:
I think `https.get` won't handle HTTP redirects, you may need to handle status codes 402/403 yourself or use a library.

https://stackoverflow.com/questions/7323932/how-do-you-follow-an-http-redirect-in-node-js